### PR TITLE
Implemented explicit version of raise_order for open knot vectors (issue #14)

### PR DESCRIPTION
**Pull request by Zetison**
To fully resolve issue #14 the case of non-open knot vectors should be addressed.

### CODE REVIEW

**Comment by TheBB** [comment]
Is it possible to move the 1D function outside somewhere? (In a util module perhaps.)

I'm not against small nested functions but this one is too big.

File: splipy/SplineObject.py [hunk 1]
```diff
@@ -438,7 +439,111 @@ def raise_order(self, *raises):
...
+            :param int m: number of degree elevations
+            :param int periodic: Number of continuous derivatives at start and end. -1 is not periodic, 0 is continuous, etc.
+            :return Q: new control points
+            """
+
+            u = np.unique(T[k-1:-k+1])
+            S = u.size - 1
+            d = P.shape[0] # dimension of spline
+
+            # Find multiplicity of the knot vector T
```
Comment (VikingScientist) [inline splipy/SplineObject.py]: See `BSplineBasis.continuity(knot)` which does this exact thing

Comment (VikingScientist) [reply splipy/SplineObject.py]:
```  python
b = BSplineBasis(k, T)
z = [k-1-b.continuity(t0) for t0 in b.knot_spans()]
```
Of course, one-line magic gives more compressed code, but often comes at the cost of readability. However this is not too bad, so I'm going to recommend it.

File: splipy/utils/__init__.py [hunk 2]
```diff
@@ -166,10 +166,76 @@ def uniquify(iterator):
...
+    :param int n: (n+1) is the number of initial basis functions
+    :param int k: spline order
+    :param T: knot vector
+    :param P: weighted NURBS coefficients
+    :param int m: number of degree elevations
+    :param int periodic: Number of continuous derivatives at start and end. -1 is not periodic, 0 is continuous, etc.
+    :return Q: new control points
+    """
+
+    u = np.unique(T[k-1:-k+1])
```
Comment (VikingScientist) [inline splipy/utils/__init__.py:181]: use `u = b.knot_spans()` (see line 187), as this is also sensitive to floating point tolerances

Comment (VikingScientist) [reply splipy/utils/__init__.py:181]: `ghost_nodes=False` should take care of `[k-1:k+1]`

**Comment by VikingScientist** [comment]
This is great. All tests passed.

**Pull request merged**